### PR TITLE
Adds a regression test for logplayback fix in #2611

### PR DESCRIPTION
**Pull request by arjo129**
Related to:

https://github.com/gazebosim/gazebo_test_cases/issues/1640

This commit adds a really simple regression check to make sure we actually publish on the right topic.

### CODE REVIEW

File: test/integration/log_system.cc [hunk 1]
```diff
@@ -774,6 +774,18 @@ TEST_F(LogSystemTest, GZ_UTILS_TEST_DISABLED_ON_WIN32(RecordAndPlayback))
   // Start server
   Server playServer(playServerConfig);
 
+  // Simulate a client
+  gz::transport::Node node;
+  std::size_t numMsgs = 0;
```
Comment (azeey) [inline test/integration/log_system.cc:779]:
```suggestion
  std::atomic<std::size_t> numMsgs = 0;
```

Since a callback is invoked from a separate thread.

**Review by azeey — APPROVED** [review]
Thanks @arjo129 ! Just one minor comment.

File: test/integration/log_system.cc (hunk 1 above)
Comment (arjo129) [reply test/integration/log_system.cc]: [c603ad6](https://github.com/gazebosim/gz-sim/pull/2619/commits/c603ad65a0b37f8a235ac37c56af7af435596fd3)

**Comment by arjo129** [comment]
https://github.com/Mergifyio backport gz-sim8

**Pull request merged**